pinot-core/src/main/java/org/apache/pinot/core/operator/filter/RangeIndexBasedFilterOperator.java

```diff
@@ -131,27 +139,75 @@ public String toExplainString() {
   }
 
   interface RangeEvaluator {
+
+    Set<FieldSpec.DataType> SUPPORTED_RAW_DATA_TYPES = EnumSet.of(FieldSpec.DataType.INT,
+            FieldSpec.DataType.LONG, FieldSpec.DataType.FLOAT, FieldSpec.DataType.DOUBLE);
+
+    static boolean canEvaluate(PredicateEvaluator predicateEvaluator, DataSource dataSource) {
+      if (dataSource.getRangeIndex() != null) {
```

Comment:
This part can be simplified. We should be able to apply the range index when:
1. Range index exist and is exact
2. Column is dictionary encoded or has supported data types
3. Predicate is EQ or RANGE (RANGE won't hit this method though)